### ISSUE TITLE
trtexec dynamic : run on onnx error

### ISSUE THREAD

**Issue by Tian14267** [NVIDIA/TensorRT#1743]
## Description

I run my onnx model with trtexec , and get failed. 
Here are the details

## Environment

**TensorRT Version**:  8.2.1
**NVIDIA GPU**:  GeForce RTX 3090
**NVIDIA Driver Version**:  Driver Version: 460.73.01 
**CUDA Version**:  11.2
**CUDNN Version**: 8.2
**Operating System**:  centos 7
**Python Version (if applicable)**:  3.6
**Tensorflow Version (if applicable)**: 1.15
**PyTorch Version (if applicable)**:  1.8.1
**Baremetal or Container (if so, version)**: 

## 1. 

My onnx has four input like this:
![image](https://user-images.githubusercontent.com/27938135/150074921-4fc13d18-7aeb-4feb-8bef-ecb57162667b.png)

## 2. 

```
./trtexec  --onnx=/data/fffan/FastSpeech.onnx \
  --minShapes=speakers:0,texts:1x2,src_lens.1:1,max_src_len:1 \
  --optShapes=speakers:1,texts:1x50,src_lens.1:50,max_src_len:50 \
  --maxShapes=speakers:10,texts:1x500,src_lens.1:500,max_src_len:500 \
  --shapes=speakers:0,texts:1x30,src_lens.1:30,max_src_len:30 \
  --workspace=4096
```

## emerging problem
```
&&&& RUNNING TensorRT.trtexec [TensorRT v8201] # /home/tensorrt_tar/TensorRT-8.2.1.8/bin/trtexec --onnx=./onnx_model/FastSpeech.onnx --minShapes=speakers:0,texts:1x2,src_lens.1:1,max_src_len:1 --optShapes=speakers:1,texts:1x50,src_lens.1:50,max_src_len:50 --maxShapes=speakers:10,texts:1x500,src_lens.1:500,max_src_len:500 --shapes=speakers:0,texts:1x30,src_lens.1:30,max_src_len:30
[01/19/2022-14:13:45] [I] === Model Options ===
[01/19/2022-14:13:45] [I] Format: ONNX
[01/19/2022-14:13:45] [I] Model: ./onnx_model/FastSpeech.onnx
[01/19/2022-14:13:45] [I] Output:
[01/19/2022-14:13:45] [I] === Build Options ===
[01/19/2022-14:13:45] [I] Max batch: explicit batch
[01/19/2022-14:13:45] [I] Workspace: 16 MiB
[01/19/2022-14:13:45] [I] minTiming: 1
[01/19/2022-14:13:45] [I] avgTiming: 8
[01/19/2022-14:13:45] [I] Precision: FP32
[01/19/2022-14:13:45] [I] Calibration: 
[01/19/2022-14:13:45] [I] Refit: Disabled
[01/19/2022-14:13:45] [I] Sparsity: Disabled
[01/19/2022-14:13:45] [I] Safe mode: Disabled
[01/19/2022-14:13:45] [I] DirectIO mode: Disabled
[01/19/2022-14:13:45] [I] Restricted mode: Disabled
[01/19/2022-14:13:45] [I] Save engine: 
[01/19/2022-14:13:45] [I] Load engine: 
[01/19/2022-14:13:45] [I] Profiling verbosity: 0
[01/19/2022-14:13:45] [I] Tactic sources: Using default tactic sources
[01/19/2022-14:13:45] [I] timingCacheMode: local
[01/19/2022-14:13:45] [I] timingCacheFile: 
[01/19/2022-14:13:45] [I] Input(s)s format: fp32:CHW
[01/19/2022-14:13:45] [I] Output(s)s format: fp32:CHW
[01/19/2022-14:13:45] [I] Input build shape: max_src_len=1+50+500
[01/19/2022-14:13:45] [I] Input build shape: src_lens.1=1+50+500
[01/19/2022-14:13:45] [I] Input build shape: speakers=0+1+10
[01/19/2022-14:13:45] [I] Input build shape: texts=1x2+1x50+1x500
[01/19/2022-14:13:45] [I] Input calibration shapes: model
[01/19/2022-14:13:45] [I] === System Options ===
[01/19/2022-14:13:45] [I] Device: 0
[01/19/2022-14:13:45] [I] DLACore: 
[01/19/2022-14:13:45] [I] Plugins:
[01/19/2022-14:13:45] [I] === Inference Options ===
[01/19/2022-14:13:45] [I] Batch: Explicit
[01/19/2022-14:13:45] [I] Input inference shape: max_src_len=30
[01/19/2022-14:13:45] [I] Input inference shape: src_lens.1=30
[01/19/2022-14:13:45] [I] Input inference shape: speakers=0
[01/19/2022-14:13:45] [I] Input inference shape: texts=1x30
[01/19/2022-14:13:45] [I] Iterations: 10
[01/19/2022-14:13:45] [I] Duration: 3s (+ 200ms warm up)
[01/19/2022-14:13:45] [I] Sleep time: 0ms
[01/19/2022-14:13:45] [I] Idle time: 0ms
[01/19/2022-14:13:45] [I] Streams: 1
[01/19/2022-14:13:45] [I] ExposeDMA: Disabled
[01/19/2022-14:13:45] [I] Data transfers: Enabled
[01/19/2022-14:13:45] [I] Spin-wait: Disabled
[01/19/2022-14:13:45] [I] Multithreading: Disabled
[01/19/2022-14:13:45] [I] CUDA Graph: Disabled
[01/19/2022-14:13:45] [I] Separate profiling: Disabled
[01/19/2022-14:13:45] [I] Time Deserialize: Disabled
[01/19/2022-14:13:45] [I] Time Refit: Disabled
[01/19/2022-14:13:45] [I] Skip inference: Disabled
[01/19/2022-14:13:45] [I] Inputs:
[01/19/2022-14:13:45] [I] === Reporting Options ===
[01/19/2022-14:13:45] [I] Verbose: Disabled
[01/19/2022-14:13:45] [I] Averages: 10 inferences
[01/19/2022-14:13:45] [I] Percentile: 99
[01/19/2022-14:13:45] [I] Dump refittable layers:Disabled
[01/19/2022-14:13:45] [I] Dump output: Disabled
[01/19/2022-14:13:45] [I] Profile: Disabled
[01/19/2022-14:13:45] [I] Export timing to JSON file: 
[01/19/2022-14:13:45] [I] Export output to JSON file: 
[01/19/2022-14:13:45] [I] Export profile to JSON file: 
[01/19/2022-14:13:45] [I] 
[01/19/2022-14:13:45] [I] === Device Information ===
[01/19/2022-14:13:45] [I] Selected Device: GeForce RTX 3090
[01/19/2022-14:13:45] [I] Compute Capability: 8.6
[01/19/2022-14:13:45] [I] SMs: 82
[01/19/2022-14:13:45] [I] Compute Clock Rate: 1.74 GHz
[01/19/2022-14:13:45] [I] Device Global Memory: 24267 MiB
[01/19/2022-14:13:45] [I] Shared Memory per SM: 100 KiB
[01/19/2022-14:13:45] [I] Memory Bus Width: 384 bits (ECC disabled)
[01/19/2022-14:13:45] [I] Memory Clock Rate: 9.751 GHz
[01/19/2022-14:13:45] [I] 
[01/19/2022-14:13:45] [I] TensorRT version: 8.2.1
[01/19/2022-14:13:46] [I] [TRT] [MemUsageChange] Init CUDA: CPU +462, GPU +0, now: CPU 473, GPU 597 (MiB)
[01/19/2022-14:13:46] [I] [TRT] [MemUsageSnapshot] Begin constructing builder kernel library: CPU 473 MiB, GPU 597 MiB
[01/19/2022-14:13:46] [I] [TRT] [MemUsageSnapshot] End constructing builder kernel library: CPU 627 MiB, GPU 641 MiB
[01/19/2022-14:13:46] [I] Start parsing network model
[01/19/2022-14:13:46] [I] [TRT] ----------------------------------------------------------------
[01/19/2022-14:13:46] [I] [TRT] Input filename:   ./onnx_model/FastSpeech.onnx
[01/19/2022-14:13:46] [I] [TRT] ONNX IR version:  0.0.7
[01/19/2022-14:13:46] [I] [TRT] Opset version:    11
[01/19/2022-14:13:46] [I] [TRT] Producer name:    pytorch
[01/19/2022-14:13:46] [I] [TRT] Producer version: 1.10
[01/19/2022-14:13:46] [I] [TRT] Domain:           
[01/19/2022-14:13:46] [I] [TRT] Model version:    0
[01/19/2022-14:13:46] [I] [TRT] Doc string:       
[01/19/2022-14:13:46] [I] [TRT] ----------------------------------------------------------------
[01/19/2022-14:13:46] [W] [TRT] onnx2trt_utils.cpp:366: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[01/19/2022-14:14:13] [W] [TRT] onnx2trt_utils.cpp:392: One or more weights outside the range of INT32 was clamped
[01/19/2022-14:14:13] [E] Error[4]: [shuffleNode.cpp::symbolicExecute::387] Error Code 4: Internal Error (Reshape_3578: IShuffleLayer applied to shape tensor must have 0 or 1 reshape dimensions: dimensions were [-1,2])
[01/19/2022-14:14:13] [E] [TRT] ModelImporter.cpp:773: While parsing node number 3589 [Pad -> "4724"]:
[01/19/2022-14:14:13] [E] [TRT] ModelImporter.cpp:774: --- Begin node ---
[01/19/2022-14:14:13] [E] [TRT] ModelImporter.cpp:775: input: "4681"
input: "4722"
input: "4723"
output: "4724"
name: "Pad_3589"
op_type: "Pad"
attribute {
  name: "mode"
  s: "constant"
  type: STRING
}

[01/19/2022-14:14:13] [E] [TRT] ModelImporter.cpp:776: --- End node ---
[01/19/2022-14:14:13] [E] [TRT] ModelImporter.cpp:779: ERROR: ModelImporter.cpp:179 In function parseGraph:
[6] Invalid Node - Pad_3589
[shuffleNode.cpp::symbolicExecute::387] Error Code 4: Internal Error (Reshape_3578: IShuffleLayer applied to shape tensor must have 0 or 1 reshape dimensions: dimensions were [-1,2])
[01/19/2022-14:14:13] [E] Failed to parse onnx file
[01/19/2022-14:14:13] [I] Finish parsing network model
[01/19/2022-14:14:13] [E] Parsing model failed
[01/19/2022-14:14:13] [E] Failed to create engine from model.
[01/19/2022-14:14:13] [E] Engine set up failed
&&&& FAILED TensorRT.trtexec [TensorRT v8201] # ./trtexec --onnx=./onnx_model/FastSpeech.onnx --minShapes=speakers:0,texts:1x2,src_lens.1:1,max_src_len:1 --optShapes=speakers:1,texts:1x50,src_lens.1:50,max_src_len:50 --maxShapes=speakers:10,texts:1x500,src_lens.1:500,max_src_len:500 --shapes=speakers:0,texts:1x30,src_lens.1:30,max_src_len:30
run_trtexec_covert.sh:行6: --workspace=4096: 未找到命令

```

**onnx model** is [Here](https://pan.baidu.com/s/1qmDTu7oqkfPzA6QVwqITTA)

**Comment by aeoleader** [comment]
@Tian14267 N-d padding is not yet supported. Check this [thread](https://github.com/NVIDIA/TensorRT/issues/1708).

**Comment by Tian14267** [comment]
> @Tian14267 N-d padding is not yet supported. Check this [thread](https://github.com/NVIDIA/TensorRT/issues/1708).

Here is an error line:

**Comment by Tian14267** [comment]
> @Tian14267 N-d padding is not yet supported. Check this [thread](https://github.com/NVIDIA/TensorRT/issues/1708).
I have two question:
**1:** 
Here is an error line:
`[01/19/2022-14:14:13] [E] Error[4]: [shuffleNode.cpp::symbolicExecute::387] Error Code 4: Internal Error (Reshape_3578: IShuffleLayer applied to shape tensor must have 0 or 1 reshape dimensions: dimensions were [-1,2])`

Does also `N-d padding` cause this error ?

**2:**
Is there any way to locate this `N-d padding` in original pytorch model ? 

Thank you very much .

**Comment by Tian14267** [comment]
@aeoleader

**Comment by aeoleader** [comment]
> > @Tian14267 N-d padding is not yet supported. Check this [thread](https://github.com/NVIDIA/TensorRT/issues/1708).
> 
> I have two question: **1:** Here is an error line: `[01/19/2022-14:14:13] [E] Error[4]: [shuffleNode.cpp::symbolicExecute::387] Error Code 4: Internal Error (Reshape_3578: IShuffleLayer applied to shape tensor must have 0 or 1 reshape dimensions: dimensions were [-1,2])`
> 
> Does also `N-d padding` cause this error ?
> 
> **2:** Is there any way to locate this `N-d padding` in original pytorch model ?
> 
> Thank you very much .

You got this error while parsing the pad node, the node name is "Pad_3589", by which you can locate the pad node in your onnx graph.

**Comment by Tian14267** [comment]
> > > @Tian14267 N-d padding is not yet supported. Check this [thread](https://github.com/NVIDIA/TensorRT/issues/1708).
> > 
> > 
> > I have two question: **1:** Here is an error line: `[01/19/2022-14:14:13] [E] Error[4]: [shuffleNode.cpp::symbolicExecute::387] Error Code 4: Internal Error (Reshape_3578: IShuffleLayer applied to shape tensor must have 0 or 1 reshape dimensions: dimensions were [-1,2])`
> > Does also `N-d padding` cause this error ?
> > **2:** Is there any way to locate this `N-d padding` in original pytorch model ?
> > Thank you very much .
> 
> You got this error while parsing the pad node, the node name is "Pad_3589", by which you can locate the pad node in your onnx graph.

OK，Thank you very much.

**Comment by ttyio** [comment]
closing issues no activity for more than 3 weeks per our policy, thanks all!
